AuthenticatorShared/Core/Platform/Services/ConfigService.swift

```diff
@@ -62,35 +109,165 @@ extension ConfigService {
+
+        let flags = FeatureFlag.debugMenuFeatureFlags.map { feature in
+            let userDefaultValue = appSettingsStore.debugFeatureFlag(name: feature.rawValue)
+            let remoteFlagValue = remoteFeatureFlags[feature]?.boolValue ?? false
+
+            return DebugMenuFeatureFlag(
+                feature: feature,
+                isEnabled: userDefaultValue ?? remoteFlagValue
+            )
+        }
+
+        return flags
+    }
+
+    func toggleDebugFeatureFlag(name: String, newValue: Bool?) async -> [DebugMenuFeatureFlag] {
+        appSettingsStore.overrideDebugFeatureFlag(
+            name: name,
+            value: newValue
+        )
+        return await getDebugFeatureFlags()
+    }
+
+    func refreshDebugFeatureFlags() async -> [DebugMenuFeatureFlag] {
+        for feature in FeatureFlag.debugMenuFeatureFlags {
+            appSettingsStore.overrideDebugFeatureFlag(
+                name: feature.rawValue,
+                value: nil
+            )
+        }
+        return await getDebugFeatureFlags()
+    }
+
+    // MARK: Private
+
+    /// Gets the server config in state depending on if the call is being done before authentication.
+    /// - Parameters:
+    ///   - config: Config to set
+    ///   - isPreAuth: If true, the call is coming before the user is authenticated or when adding a new account
+    func getStateServerConfig(isPreAuth: Bool) async throws -> ServerConfig? {
+        guard !isPreAuth else {
+            return await stateService.getPreAuthServerConfig()
+        }
+        return try? await stateService.getServerConfig()
+    }
+
+    func configPublisher() async throws -> AsyncThrowingPublisher<AnyPublisher<MetaServerConfig?, Never>> {
+        configSubject.eraseToAnyPublisher().values
+    }
+
+    /// Sets the server config in state depending on if the call is being done before authentication.
+    /// - Parameters:
+    ///   - config: Config to set
+    ///   - isPreAuth: If true, the call is coming before the user is authenticated or when adding a new account
+    ///   - userId: The userId to set the server config to.
+    func setStateServerConfig(_ config: ServerConfig, isPreAuth: Bool, userId: String? = nil) async throws {
+        guard !isPreAuth else {
+            await stateService.setPreAuthServerConfig(config: config)
+            return
+        }
+        try? await stateService.setServerConfig(config, userId: userId)
+    }
+}
+
+/// Helper object to send updated server config object with extra metadata
+/// like whether it comes from pre-auth and the user ID it belongs to.
+/// This is useful for getting the config on background and establishing which was the original context.
+struct MetaServerConfig {
+    /// If true, the call is coming before the user is authenticated or when adding a new account
+    let isPreAuth: Bool
+    /// The user ID the requested the server config.
+    let userId: String?
+    /// The server config.
+    let serverConfig: ServerConfig?
 }
```

Comment:
🤔 This alongside the subject/publisher make sense if we have the `updateConfigFromServer` method and the API call, so not sure if it's worth adding it here now given that it won't ever be used until the API layer is added.